packages/grafana-data/src/transformations/transformers/fillGaps.test.ts

```diff
@@ -0,0 +1,71 @@
+import { FieldType, MutableDataFrame } from '@grafana/data';
+import { fillGaps } from './fillGaps';
+
+describe('fillGaps Transformer', () => {
+  test('should fill nulls as expected', () => {
+    const df = new MutableDataFrame({
+      refId: 'A',
+      fields: [
+        { name: 'Time', type: FieldType.time, values: [1, 3, 10] },
+        { name: 'One', type: FieldType.number, config: { interval: 1 }, values: [4, 6, 8] },
```

Comment:
this is a bit strange/surprising. it uses the interval of a non-time sibling field to increment the time field by that interval? i'm not sure if this is necessary (or even desirable).

is there a specific scenario where this might occur?

Reply:
to me it's a bit confusing as well. do we need it on each field, or should this be switched to the frame?

if it's on the frame, my thought here was that the transformation would just pick the interval from the first field that has it set for a couple reasons:
* using field config on the backend causes extra allocations, and in the case of prometheus we are already setting config options on the value field, so it seemed like setting the interval on the value field made sense in that case.
* it's unclear to me what happens to field configs when other transformations are applied and fields are merged

in general though i agree that it probably makes more sense on the time field, and i can restrict it to that field if that's the preferred route.

Reply:
@leeoniya after talking with @ryantxu about this one, i made an incorrect assumption that this would only apply to time fields in the first version, so ignore my comment above